django/db/models/base.py

```diff
@@ -513,18 +513,21 @@ def __init__(self, *args, **kwargs):
 
         if kwargs:
             property_names = opts._property_names
-            for prop in tuple(kwargs):
+            unexpected = []
+            for prop, value in kwargs.items():
                 try:
                     # Any remaining kwargs must correspond to properties or
                     # virtual fields.
                     if prop in property_names or opts.get_field(prop):
-                        if kwargs[prop] is not _DEFERRED:
-                            _setattr(self, prop, kwargs[prop])
-                        del kwargs[prop]
+                        if value is not _DEFERRED:
+                            _setattr(self, prop, value)
                 except (AttributeError, FieldDoesNotExist):
-                    pass
-            for kwarg in kwargs:
-                raise TypeError("%s() got an unexpected keyword argument '%s'" % (cls.__name__, kwarg))
+                    unexpected.append(prop)
```

Comment:
Do we need to catch `AttributeError`? It looks unnecessary since d2a26c1a90e837777dabdf3d67ceec4d2a70fb86 :thinking:

Reply:
I believe you're right - it could only occur from properties, which are already handled.

Reply:
Dropping `AttributeError` causes `properties.tests.PropertyTests.test_setter` to fail, because it expects the `TypeError` instead of `AttributeError`. IMO allowing the `AttributeError` to be raised is fine, it highlights that the property is the problem. Changing the test.